### DISCUSSION TITLE
No compatible device found

### DISCUSSION THREAD

**Discussion by jeagh** [pybricks/support#884] [Q&A] [answered]
I'm trying to install the latest pybricks firmware on my Mindstorms hub and I've followed the instructions to the letter three times. The installation finishes with the hub's large button blinking blue and there is a 3*3 LED display. Even though the installation seems to have succeeded I cannot establish a Bluetooth connection. The dialog box displays "No compatible device found."

I've spent hours searching for any clues on this site and the Internet in general but I'm still no wiser. I must be missing something that should be obvious.

I've tried both Chromium and Chrome under Linux and Chrome Windows and have connected to "beta.pybricks.com".

Any hints will be gratefully appreciated. 

I just discovered that my Android phone does see the pybricks hub under Bluetooth devices so I wonder if my laptop may be too old? The Lego Mindstorms app does work correctly under Windows on the same laptop which I can boot into Linux or Windows.

**Comment by dlech** [comment, marked as answer]
The official LEGO app uses Bluetooth classic but Pybricks uses Bluetooth low energy. Is is possible that the Bluetooth adapter on your computer doesn't support Bluetooth low energy? Also, on Linux only, you have to enable experimental features in the Chromium browser.

You could also try scanning with `chrome://bluetooth-internals` or an app like [this](https://apps.microsoft.com/store/detail/bluetooth-le-explorer/9N0ZTKF1QD98?hl=en-us&gl=us) on Windows to see if you can see the hub at all when it is advertising.

**Reply by jeagh** [reply]
Thank you for your reply, I didn't really expect a reply on Christmas Day.

I wonder if I can buy a Bluetooth low energy USB adapter for my laptop and will it be supported by the OS?

I do have have experimental features enabled and I will try bluetoth-internals.

I can use pybricks on my Android tablet but it is a pain without a keyboard.

**Reply by dlech** [reply]
USB Bluetooth adapters can work, but seem to be hit or miss on Windows. I've had luck with ones labeled as "CSR 4.0".

**Comment by jeagh** [comment]
I notice that I wrote tablet, I meant phone. I did try my Android tablet but pressing the Bluetooth button brings up an error message that tells me to use Chrome or enable experimental features. I am using Chrome and I do have experimental features enabled. The tablet does see the pybricks hub. There must be some difference between the Chrome versions on the tablet and phone.

**Reply by dlech** [reply]
Experimental features only needs to be enabled on Linux. No other operating system needs this. What is the version of Chrome that is not working?

**Reply by jeagh** [reply]
Thanks for your reply. I have the latest version of Chrome on the phone and tablet. The phone originally worked without enabling experimental features but now, just like the table, I get the error message "use Chrome etc".  I can sometimes establish a Bluetooth connection with the phone via beta.pybricks.com (I've also tried pybricks.com) but I have never been able to get the tablet to connect. The phone is unusable without a keyboard anyway. I will disable the experimental features.

I will order a Bluetooth adaptor for my laptop.

**Reply by dlech** [reply]
Latest version is "Chrome 108.0.5359.128"? Is Bluetooth turned on on the tablet? Does the Chrome app have permission to use Bluetooth?

**Reply by jeagh** [reply]
Thank you again dlech. Just in case someone else has the same problem the answer is "enable web bluetooth new permissions backend".

Now to see if I can use the colour sensor in my project.

**Comment by gingergrinch** [comment]
Hello ,

I have a similar probelm:

Trying to connect a technic hub with pybricks always fails, because no device is found.
The adapter overview in chromium looks like this:

![image](https://user-images.githubusercontent.com/123868934/215320798-6b5a1944-87e2-423d-b620-4d5b3bf90be1.png)

The technic hub can be found and it is also possible to pair it in chromium:

![image](https://user-images.githubusercontent.com/123868934/215320850-3c0d7532-0dc0-4309-9bda-cd02ad3e1640.png)

![image](https://user-images.githubusercontent.com/123868934/215320871-24275d71-a841-4c04-8800-1c17d5d9e50c.png)

The laptop has Bluetooth 4.1.
I have also tried adding several registry entries in terms of group policies etc., but no success.
Also an other device with BT 5.0 did not get any search result in the pybricks app. :-(

The hub is working in general as it is possible to connect it with the original Lego technic app.
I have also tried with a second laptop, but also without success.

I would appreciate any tips!
Thanks!

**Reply by jeagh** [reply]
I was hoping that someone more knowledgeable than me would answer. The only advice that I can offer is to make sure that "#enable-experimental-web-platform-features" and "#enable-web-bluetooth-new-permissions-backend" have been enabled in the browser. My laptop does not have Bluetooth Low Energy and so I have bought a BT 5 adaptor which works perfectly under Linux and Windows 10.
